src/every_query/train.py

```
@@ -82,16 +83,33 @@ def get_step(fp: Path) -> int:
 
-    task_str = f"{'|'.join(sorted(cfg.query.codes))}_{'|'.join(str(d) for d in sorted(durations))}"
+    geometric_p = cfg.query.geometric_p
+    max_query_codes = cfg.query.max_query_codes
+    seed = cfg.get("seed", 1)
+    num_queries = cfg.query.queries_per_shard
+
+    task_str = (
+        f"{'|'.join(sorted(cfg.query.codes))}_{'|'.join(str(d) for d in sorted(durations))}"
+        f"_gp{geometric_p}_mqc{max_query_codes}_qps{num_queries}_seed{seed}"
+    )
     hash_hex = hashlib.md5(task_str.encode()).hexdigest()
```

Ruff (0.15.7)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)